msu/systems/mod_settings/abstract_setting.nut

```diff
@@ -122,13 +124,13 @@
 
 	function getLockReason()
 	{
-		return this.LockReason;
+		return this.LockReason == "" ? "" : this.LockReason.slice(0, -5);
 	}
 
 	function lock( _lockReason = "" )
 	{
 		this.Locked = true;
-		if (_lockReason != "") this.LockReason = _lockReason;
+		if (_lockReason != "") this.LockReason += _lockReason + " +++ ";
```

Comment:
Should the reason be optional?

Reply:
On second thought, I think yeah the reason shouldn't be optional. But this is not changeable anymore as this is a released feature. @Enduriel What are your thoughts? Is there the possibility to deprecate and create a new function with reason not being optional?

Reply:
Actually one way to achieve this would be to add a `lockSetting` function to `ModSettingsModAddon` which modders must use to lock other mod's settings. That way we can pass the locker's id which also fixes the issue that the locker's mod id is always printed in lock reason.

Currently people lock other mod's settings by directly accessing that setting and locking it. This isn't good.

Reply:
> Currently people lock other mod's settings by directly accessing that setting and locking it. This isn't good.

I'm not sure I agree with this, I think that's fine.

I think this is overcomplicating the issue, if someone is locking a setting, they presumably know what they're doing, I don't think we should be trying to block/handhold to this extent.